### PR TITLE
net, doc: update I2P hardcoded seeds and docs for 22.0

### PR DESCRIPTION
**Pull request by jonatack**
- Update the I2P seeds with the latest ones I can connect to (feedback welcome).

- Update the I2P doc with frequently asked and reported info: IBD speed and running with other networks.

### CODE REVIEW

**Comment by jonatack** [comment]
These I2P addresses can be somewhat intermittent. A few other addresses that seem to be less-often connectable recently:

```
22plzy5mqwz64iukd2hbsa5oo454iwkjko26tifi54ir4mbnpfea.b32.i2p
72l3ucjkuscrbiiepoehuwqgknyzgo7zuix5ty4puwrkyhtmnsga.b32.i2p
pjs7or2ctvteeo5tu4bwyrtydeuhqhvdprtujn4daxr75jpebjxa.b32.i2p
sedndhv5vpcgdmykyi5st4yqhdxl3hpdtglta4do435wupahhx6q.b32.i2p
vnqix5uljfwssvzbexzjeuoehjujtw4iqftiu34nbvig64kkmnla.b32.i2p
wxgpc5mtv5dk2kzpgqz7ybjtmumrnavfoa36o4mas4ksc3owqgya.b32.i2p
```

Feedback and additional addresses welcome.

**Comment by jonatack** [comment]
Some of the I2P addresses can be somewhat intermittent. 

A few other addresses I have that seem to be less-often connectable recently:

```
22plzy5mqwz64iukd2hbsa5oo454iwkjko26tifi54ir4mbnpfea.b32.i2p
72l3ucjkuscrbiiepoehuwqgknyzgo7zuix5ty4puwrkyhtmnsga.b32.i2p
pjs7or2ctvteeo5tu4bwyrtydeuhqhvdprtujn4daxr75jpebjxa.b32.i2p
sedndhv5vpcgdmykyi5st4yqhdxl3hpdtglta4do435wupahhx6q.b32.i2p
vnqix5uljfwssvzbexzjeuoehjujtw4iqftiu34nbvig64kkmnla.b32.i2p
wxgpc5mtv5dk2kzpgqz7ybjtmumrnavfoa36o4mas4ksc3owqgya.b32.i2p
```

Feedback and additional addresses welcome.

**Comment by jonatack** [comment]
(Don't merge yet. People are replying offline regarding their I2P nodes, will update.)

**Review by vasild — APPROVED** [review]
ACK d2dffd5be4c8f6a1942dd971d09707c3620a1689

No changes by:
```sh
contrib/seeds/generate-seeds.py contrib/seeds > src/chainparamsseeds.h
```
(so, `src/chainparamsseeds.h` has been generated from `contrib/seeds` indeed).

I tried to connect to all I2P nodes from the seeds and compared the seeds with the nodes I am currently connected to:
```sh
#  get I2P seeds
grep "b32\.i2p" contrib/seeds/nodes_main.txt > /tmp/seeds

# try to connect to all of them
for p in $(cat /tmp/seeds) ; do bitcoin-cli addnode $p onetry ; done

# see who is online (include our own addr)
( bitcoin-cli getpeerinfo |jq -r 'map(select(.network == "i2p")) |.[] |.addr' ; bitcoin-cli getnetworkinfo |jq -r '.localaddresses |map(select(.address |test("\\.i2p$"; "i"))) |.[] |.address + ":0"') |sort > /tmp/online

# compare
diff -u0 /tmp/seeds /tmp/online
```

result:
```diff
--- /tmp/seeds	2021-07-30 14:32:26.250310000 +0200
+++ /tmp/online	2021-07-30 14:30:56.333247000 +0200
@@ -5,2 +4,0 @@
-c4gfnttsuwqomiygupdqqqyy5y5emnk5c73hrfvatri67prd7vyq.b32.i2p:0
-dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p:0
@@ -13,0 +12 @@
+sedndhv5vpcgdmykyi5st4yqhdxl3hpdtglta4do435wupahhx6q.b32.i2p:0
```

* I connect successfully to `c4gf...` but get disconnected soon after. Maybe that node is overloaded.

* `dhtq...` gives me `Unexpected reply to "NAMING LOOKUP NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p": "NAMING REPLY RESULT=KEY_NOT_FOUND NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p"`

* `sedn...` is an online node, not present in the seeds

**Comment by jonatack** [comment]
>     * I connect successfully to `c4gf...` but get disconnected soon after. Maybe that node is overloaded.

Agree, that node can be selective but has been stable since February.

>     * `dhtq...` gives me `Unexpected reply to "NAMING LOOKUP NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p": "NAMING REPLY RESULT=KEY_NOT_FOUND NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p"`

It was stable until very recently. Sent a message and am waiting to hear from its operator. 

>     * `sedn...` is an online node, not present in the seeds

It was offline for a while. Sent a message and its operator brought it back up and says will maintain its stability.  I'll add it in the next push.

**Comment by tryphe** [comment]
ACK d2dffd5be4c8f6a1942dd971d09707c3620a1689

It's funny how close this is to my list of I2P peers.

Here's a few more I've seen in the past few days that you can add at your discretion:
```
raoxmlv76s4f4uu275uzaorlr6f2m3yojbspwiciitz6mnl5qnda.b32.i2p:0
igaqpxjupsix5vpxt2bk5dql3md55hvk2n65jzpkwbw6juogyskq.b32.i2p:0
zhjaqw53flrpjvic3ieihzhhkmxedkoccynhowongdkviivtxbmq.b32.i2p:0
```

**Comment by duncandean** [comment]
ACK d2dffd5

I don't have any additional addresses I've seen that haven't already been mentioned here.

> I connect successfully to c4gf... but get disconnected soon after. Maybe that node is overloaded.

I am currently connected to `c4gfnttsuwqomiygupdqqqyy5y5emnk5c73hrfvatri67prd7vyq.b32.i2p:0` and it has been stable for me for a while already.

**Comment by jonatack** [comment]
> Here's a few more I've seen in the past few days that you can add at your discretion:
> 
> ```
> raoxmlv76s4f4uu275uzaorlr6f2m3yojbspwiciitz6mnl5qnda.b32.i2p:0
> igaqpxjupsix5vpxt2bk5dql3md55hvk2n65jzpkwbw6juogyskq.b32.i2p:0
> ```

Yes, these have been stable over the day today. I'm waiting on a couple of people to finish updating their I2P nodes before pushing an update.

**Comment by willcl-ark** [comment]
ACK d2dffd5be4c8f6a1942dd971d09707c3620a1689

I am connected to about 8 of these, with the hardcoded ones pretty much being the most stable 👍🏼

**Comment by laanwj** [comment]
ACK d2dffd5be4c8f6a1942dd971d09707c3620a1689

**Comment by fanquake** [comment]
Backported in #22534.

**Comment by jonatack** [comment]
> > `dhtq...` gives me `Unexpected reply to "NAMING LOOKUP NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p": "NAMING REPLY RESULT=KEY_NOT_FOUND NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p"`
> 
> It was stable until very recently. Sent a message and am waiting to hear from its operator.
> 

Heard back and its back up.

**Comment by jonatack** [comment]
> > `dhtq...` gives me `Unexpected reply to "NAMING LOOKUP NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p": "NAMING REPLY RESULT=KEY_NOT_FOUND NAME=dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p"`
> 
> It was stable until very recently. Sent a message and am waiting to hear from its operator.
> 

Heard back and `dhtq2p76tyhi442aidb3vd2bv7yxxjuddpb2jydnnrl2ons5bhha.b32.i2p` is back online... i2pd had stopped working (ubuntu 20.04, i2pd 2.38.0).

File: doc/i2p.md [hunk 1]
```diff
@@ -47,13 +47,21 @@ information in the debug log about your I2P configuration and connections. Run
...
-`onlynet=i2p`. I2P support was added to Bitcoin Core in version 22.0 (mid 2021)
+`onlynet=i2p`. I2P support was added to Bitcoin Core in version 22.0 (mid-2021)
 and there may be fewer I2P peers than Tor or IP ones. Therefore, using
 `onlynet=i2p` alone (without other `onlynet=`) may make a node more susceptible
 to [Sybil attacks](https://en.bitcoin.it/wiki/Weaknesses#Sybil_attack). Use
 `bitcoin-cli -addrinfo` to see the number of I2P addresses known to your node.
 
-## I2P related information in Bitcoin Core
+Another consideration with `onlynet=i2p` is that the initial blocks download
+phase when syncing up a new node can be very slow. This phase can be sped up by
```
Comment (gruve-p) [inline doc/i2p.md:57]: sped = speed

Comment (jonatack) [reply doc/i2p.md:57]: "sped" is the past tense

Comment (jonatack) [reply doc/i2p.md:57]: https://www.merriam-webster.com/thesaurus/sped%20%28up%29

**Pull request merged**